### PR TITLE
Add workflow for publishing on PyPI and TestPyPI

### PR DESCRIPTION
**Pull request by DanielKotik**
I've added secrets `TEST_PYPI_API_TOKEN` and `PYPI_API_TOKEN` to the mala settings.

On pushes to `master` and `develop` this workflow gets triggered and tries to build the mala package. On tags being pushed, the package will also be uploaded to TestPyPI. Only when a tag is turned into a release via GitHub, the package is published on PyPI.

### CODE REVIEW

**Comment by DanielKotik** [comment]
@RandomDefaultUser Please change `setup(name='mala', ...)` in `setup.py` to a reasonable name that is not already taken on PyPI. Any suggestions?

**Comment by RandomDefaultUser** [comment]
Some suggestions for the name:

1. `mala-core`: it may be that we eventually build more and more software around the `mala` ecosystem, e.g. an additional automation layer for HPC, so this would be the "core" package
2. `mala-dft`: since we mostly work with DFT data; technically that's not exclusive, I am aware of that. 
3. `mala-ml`: since we work with machine learning, and the other `mala` package does not
4. `mala-ms`: since we are concerned with materials science
5. `materials-learning-algorithms`: this may be excessively long

I would advocate for 1.

**Comment by Zevrap-81** [comment]
if you want something that sounds good then maybe "pymala"?

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ This repository is structured as follows:
 
 ## Installation
 
+> **WARNING**: Even if you install MALA via PyPI, please consult the full installation instructions afterwards. External modules (like the QuantumESPRESSO bindings) are not distributed via PyPI!
```
Comment (DanielKotik) [inline README.md:26]: Is this `>` before `**WARNING**` of any value, or just a typo?

**Review by DanielKotik — COMMENTED** [review]
see comment

File: README.md (hunk 1 above)
Comment (RandomDefaultUser) [reply README.md:26]: It's a blockquote, per this suggestion: https://stackoverflow.com/a/55850542

Comment (DanielKotik) [reply README.md:26]: You're right.

**Pull request merged**